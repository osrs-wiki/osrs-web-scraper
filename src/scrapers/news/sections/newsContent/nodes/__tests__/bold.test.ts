import { MediaWikiBuilder } from "@osrs-wiki/mediawiki-builder";
import parse from "node-html-parser";

import boldParser from "../bold";

describe("bold node", () => {
  test("Bold text should parse and render", () => {
    const root = parse("<b>test</b>");
    const builder = new MediaWikiBuilder();
    builder.addContents([boldParser(root.firstChild)].flat());
    expect(builder.build()).toMatchSnapshot();
  });

  test("Bold text should parse and render with a link", () => {
    const root = parse(
      "<b>testing this <a href='https://example.com'>link</a></b>"
    );
    const builder = new MediaWikiBuilder();
    builder.addContents([boldParser(root.firstChild)].flat());
    expect(builder.build()).toMatchSnapshot();
  });
  test("Bold text should parse and render with a link and italics", () => {
    const root = parse(
      "<b>testing this <a href='https://example.com'>link</a> and <i>italics</i></b>"
    );
    const builder = new MediaWikiBuilder();
    builder.addContents([boldParser(root.firstChild)].flat());
    expect(builder.build()).toMatchSnapshot();
  });
});
