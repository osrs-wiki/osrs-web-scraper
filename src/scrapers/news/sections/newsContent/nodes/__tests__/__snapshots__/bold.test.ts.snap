// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bold node Bold text should parse and render 1`] = `"'''test'''"`;

exports[`bold node Bold text should parse and render with a link 1`] = `"'''testing this [https://example.com link]'''"`;

exports[`bold node Bold text should parse and render with a link and italics 1`] = `"'''testing this [https://example.com link] and ''italics'''''"`;
